### PR TITLE
Fix view detach order

### PR DESCRIPTION
**Pull request by zsoltk**

### CODE REVIEW

**Comment by zsoltk** [comment]
The newly added check caught a problem: if the last entry in back stack is Overlay, the view of the previous entry is not removed at this point (which is expected). But view detach calls cleanup only on the last element's RoutingAction (which is the Overlay), while it should call it on the last content-type entry + all its overlays. Not a serious issue at this point, but needs to be solved anyway.

**Comment by zsoltk** [comment]
Should be retried once #53 is merged.

**Comment by zsoltk** [comment]
Works after rebase.

File: android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt [hunk 1]
```diff
@@ -193,6 +194,14 @@ open class Node<V : RibView>(
     }
 
     open fun onDetach() {
+        if (isViewAttached) {
```
Comment (lukaville) [inline android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt]: can we cover this with a test?

Comment (zsoltk) [reply android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt]: It is covered in `ConfigurationFeatureTest`, look for test cases containing "inOrder".

Comment (lukaville) [reply android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt]: I mean the logic that automatically detaches the view and reports a non-fatal error if the view is not detached before `Node.onDetach()`

Comment (zsoltk) [reply android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt]: Done.

**Pull request merged**